packages/destination-actions/src/destinations/engage/utils/apiLookups.ts

```
@@ -137,51 +146,60 @@ export const performApiLookup = async (
       data = await res.data
     } catch (error: any) {
       const respError = error.response as ResponseError
-      logger?.error(`TE Messaging: Email api lookup failure - api lookup id: ${id} - ${settings.spaceId} - [${error}]`)
+      logDataFeedError('Data feed call failure', error)
       datafeedTags.push(
         `error:true`,
         `error_message:${error.message}`,
         `error_status:${respError?.status}`,
         'reason:api_call_failure'
       )
-      // Rethrow error to preserve default http retry logic
-      throw error
+
+      // If retry is enabled for this data feed then rethrow the error to preserve centrifuge default retry logic
+      if (shouldRetryOnRetryableError) {
+        throw error
```

Are we mandating that this be retried? If so, we might want to wrap the error in this https://github.com/segmentio/action-destinations/blob/28ae9c85e9f9c5ff9af12b8a007b28f54a14d7a1/packages/core/src/errors.ts#L49

---

Not mandated for it to retry.  I was trying to communicate that with `onRetryableError` part of `shouldRetryOnRetryableError`.  I think the naming of that can be improved.

We purposefully rethrow the http error so that we follow normal centrifuge retry/fail logic based on the error status. In the other case, we never want to retry, so just ignore the error and return empty data for the data feed.

appreciate the review!  thanks!